src/Field/WfsSearchField/WfsSearchField.tsx

```diff
@@ -0,0 +1,256 @@
+import './WfsSearchField.less';
+
+import { faCircleNotch, faClose } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import Logger from '@terrestris/base-util/dist/Logger';
+import { SearchConfig } from '@terrestris/ol-util/dist/WfsFilterUtil/WfsFilterUtil';
+import useMap from '@terrestris/react-util/dist/Hooks/useMap/useMap';
+import { useWfs } from '@terrestris/react-util/dist/Hooks/useWfs/useWfs';
+import { AutoComplete, Input, Spin } from 'antd';
+import { DefaultOptionType, OptionProps } from 'antd/lib/select';
+import _get from 'lodash/get';
+import _isNil from 'lodash/isNil';
+import _isString from 'lodash/isString';
+import OlFeature from 'ol/Feature';
+import React, { FC, useEffect, useMemo, useState } from 'react';
+
+import { CSS_PREFIX } from '../../constants';
+
+const Option = AutoComplete.Option;
+
+export interface WfsSearchState {
+  data: Array<any>;
+  fetching: boolean;
+  searchTerm: string;
+}
+
+export type WfsSearchFieldProps = {
+  additionalFetchOptions?: Partial<RequestInit>;
+  asInput?: boolean;
+  baseUrl: string;
+  className?: string;
+  displayValue?: string;
+  idProperty?: string;
+  minChars?: number;
+  onBeforeSearch?: (value: string) => string;
+  onChange?: (val: OlFeature[] | undefined) => undefined;
+  value?: OlFeature[] | undefined;
+} & SearchConfig;
+
+const defaultClassName = `${CSS_PREFIX}wfssearch`;
+
+/**
+ * The WfsSearchField field.
+ * Implements an input field to do a WFS-GetFeature request.
+ *
+ * The GetFeature request is created with `ol.format.WFS.writeGetFeature`
+ * so most of the WFS specific options work like document in the corresponding
+ * API-docs: https://openlayers.org/en/latest/apidoc/module-ol_format_WFS.html
+ *
+ * @class WfsSearchField
+ * @extends React.Component
```

Comment:
(very minor) I think these annotations are meant for class components. I would just remove them since we use typescript anyway

Reply:
```suggestion
```